src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -457,11 +477,52 @@ export class MeetingManager implements AudioVideoObserver {
         this.selectAudioInputDeviceError = error;
         console.error('Failed to choose audio input device.', error);
       }
-      this.selectedAudioInputDevice = deviceId;
+  
+      let innerDevice = null;
+      if (isAudioTransformDevice(device)) {
+        innerDevice = await device.intrinsicDevice();
+      } else {
+        innerDevice = device;
+      }
+
+      if (is440) {
+        this.selectedAudioInputDevice = '440';
+      } else if (innerDevice === null) {
+        this.selectedAudioInputDevice = null;
+      } else if (typeof innerDevice === 'string') {
+        this.selectedAudioInputDevice = innerDevice;
+      } else if (innerDevice instanceof MediaStream) {
+        this.selectedAudioInputDevice = innerDevice.id;
+      } else {
+        const deviceId = this.getDeviceID(innerDevice);
+        this.selectedAudioInputDevice = deviceId;
+      }
+
+      this.selectedAudioInputTransformDevice = device;
     }
     this.publishSelectedAudioInputDevice();
     this.publishSelectAudioInputDeviceError();
-  };
+    this.publishSelectedAudioInputTransformDevice();
+  };  
+
+  private getDeviceID = (innerDevice: MediaTrackConstraints| MediaTrackConstraintSet): string => {
+    if (Array.isArray(innerDevice)) {
+      innerDevice = innerDevice[0];
+    }
+
+    const deviceId = innerDevice.deviceId;
+    if (typeof(deviceId) === 'string') {
+      return deviceId;
+    } else if (Array.isArray(deviceId)) {
+      return deviceId[0];
```

Comment:
No `else` after `return`.

Reply:
Do you mean remove the last `else` and just return `''` after if statement?

Reply:
```typescript
if (foo) {
  return bar;
}
if (other) {
  return baz;
}
…
```

Reply:
Thanks, updated!